From a77b267abb870c935f9b4187ca31409fa5eb9dcb Mon Sep 17 00:00:00 2001
From: Peter Wu <lekensteyn@gmail.com>
Date: Thu, 26 Sep 2013 00:55:55 +0200
Subject: [PATCH] Fix compile warning about unused typedef

Switch assertion to style as used by the Linux kernel (BUILD_BUG_ON)

Signed-off-by: Peter Wu <lekensteyn@gmail.com>
---
 lens_default.c | 2 +-
 value.c        | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lens_default.c b/lens_default.c
index e0c0566..84b74ea 100644
--- a/lens_default.c
+++ b/lens_default.c
@@ -345,7 +345,7 @@ format_array(FILE *stream, struct value *value, struct value_dict *arguments,
 {
 	/* We need "long" to be long enough to cover the whole address
 	 * space.  */
-	typedef char assert__long_enough_long[-(sizeof(long) < sizeof(void *))];
+	(void)sizeof(char[1 - 2*(sizeof(long) < sizeof(void *))]);
 	long l;
 	if (expr_eval_word(length, value, arguments, &l) < 0)
 		return -1;
diff --git a/value.c b/value.c
index d18db17..8caf98c 100644
--- a/value.c
+++ b/value.c
@@ -284,7 +284,7 @@ value_init_deref(struct value *ret_val, struct value *valp)
 
 	/* We need "long" to be long enough to hold platform
 	 * pointers.  */
-	typedef char assert__long_enough_long[-(sizeof(l) < sizeof(void *))];
+	(void)sizeof(char[1 - 2*(sizeof(l) < sizeof(void *))]);
 
 	value_common_init(ret_val, valp->inferior, valp,
 			  valp->type->u.ptr_info.info, 0);
-- 
1.8.5.2

